### PR TITLE
feat(npm): npmrcMerge

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds new feature `npmrcMerge`.

## Context:

Provide flexibility so that `config.npmrc` doesn't *need* to override `.npmrc` in the repo.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1232,6 +1232,13 @@ Typically you would encrypt it and put it inside the `encrypted` object.
 
 See [Private npm module support](https://docs.renovatebot.com/getting-started/private-packages) for details on how this is used.
 
+## npmrcMerge
+
+This option exists to provide flexibility about whether `npmrc` strings in config should override `.npmrc` files in the repo, or be merged with them.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1237]: Can you explain in the text why you might need/want to use this?

It's not clear to me what the possible benefit is to using/not using this option... 😉

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 1 above)
Comment (rarkins) [reply docs/usage/configuration-options.md:1237]: See changes

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1232,6 +1232,14 @@ Typically you would encrypt it and put it inside the `encrypted` object.
...
+In some situations you need the ability to force override `.npmrc` contents in a repo (`npmMerge=false`) while in others you might want to simply supplement the settings already in the `.npmrc` (`npmMerge=true`). A use case for the latter is if you are a Renovate bot admin and wish to provide a default token for `npmjs.org` without removing any other `.npmrc` settings which individual repositories have configured (such as scopes/registries).
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1238]:
```suggestion
In some situations you need the ability to force override `.npmrc` contents in a repo (`npmMerge=false`) while in others you might want to simply supplement the settings already in the `.npmrc` (`npmMerge=true`).
A use case for the latter is if you are a Renovate bot admin and wish to provide a default token for `npmjs.org` without removing any other `.npmrc` settings which individual repositories have configured (such as scopes/registries).
```

Let's use one sentence per line here. 😄

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -589,6 +589,14 @@ const options: RenovateOptions[] = [
     stage: 'branch',
     type: 'string',
   },
+  {
+    name: 'npmrcMerge',
+    description:
+      'Whether to merge config npmrc with repo .npmrc content if both are found.',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:595]:
```suggestion
      'Whether to merge config.npmrc with repo .npmrc content if both are found.',
```

Not sure which is correct, so please check this suggestion. 😉

**Review by HonkingGoose — APPROVED** [review]
I think the docs are okay now.

It's still a bit tricky for me to understand, but I think that's because I don't use this kind of stuff... 😄

**Review by viceice — APPROVED** [review]

**Pull request merged**